lib/Backend/IR.cpp

```diff
@@ -1187,12 +1186,17 @@ Instr::ReplaceBailOutInfo(BailOutInfo *newBailOutInfo)
     if (oldBailOutInfo->bailOutInstr == this)
     {
         JitArenaAllocator * alloc = this->m_func->m_alloc;
+        // If the oldBailOutInfo's captured values were cached on the globopt-block-data,
+        // make a deep copy of the lists in the captured values on the block data.
+        if (block && block->globOptData.capturedValuesCandidate == &oldBailOutInfo->capturedValues)
+        {
+            block->globOptData.capturedValuesCandidate = oldBailOutInfo->capturedValues.Clone(block->globOptData.GetGlobOptAllocator());
```

Comment:
Do we really have to clone in this scenario ?
It sounds like we have 2 references to the same arena allocated bailout info.
Shouldn't we just not delete it and leave a reference in GlobOptData?

In general, do we really have to free those bailout info ?
Whenever memory management gets somewhat complicated, shouldn't we just let the arena clean things up once we're done jitting that function ?

Reply:
Makes sense. I'll change it up.